### PR TITLE
Fix to #18002 - Query/Test: convert AssertQuery methods to strongly typed versions (part2)

### PR DESCRIPTION
**Pull request by maumar**
converting AssertQuery(...) (i.e. queryable, non scalar result) to strongly typed versions

### CODE REVIEW

File: test/EFCore.Cosmos.FunctionalTests/Query/SimpleQueryCosmosTest.cs [hunk 1]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.Linq;
 using System.Linq.Expressions;
+using System.Runtime.Intrinsics.X86;
```
Comment (smitpatel) [inline test/EFCore.Cosmos.FunctionalTests/Query/SimpleQueryCosmosTest.cs]: wow!

File: test/EFCore.Specification.Tests/Query/GroupByQueryTestBase.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Linq;
+using System.Runtime.Intrinsics.X86;
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/GroupByQueryTestBase.cs]: Verify all usings

**Comment by smitpatel** [comment]
>         public override async Task AssertQueryTyped<TResult>(

Naming?

---
Refers to: test/EFCore.Specification.Tests/TestUtilities/QueryAsserter.cs:302 in cb9a7c0. [](commit_id = cb9a7c02ae9bc02753de6ffad4a14ef37bd130fb, deletion_comment = False)

File: test/EFCore.SqlServer.FunctionalTests/Query/GearsOfWarQuerySqlServerTest.cs [hunk 3]
```diff
@@ -4288,68 +4293,71 @@ public override async Task Correlated_collections_project_anonymous_collection_r
...
-    LEFT JOIN (
-        SELECT [s1].[SquadId], [s1].[MissionId]
-        FROM [SquadMissions] AS [s1]
-        WHERE [s1].[SquadId] < 7
-    ) AS [t] ON [m].[Id] = [t].[MissionId]
-    WHERE [s0].[MissionId] < 42
-) AS [t0] ON [s].[Id] = [t0].[SquadId]
-ORDER BY [s].[Id], [t0].[SquadId], [t0].[MissionId], [t0].[Id], [t0].[SquadId0], [t0].[MissionId0]");
-        }
+        // issue #18002
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/Query/GearsOfWarQuerySqlServerTest.cs]: Why do we need to do this?

Comment (maumar) [reply test/EFCore.SqlServer.FunctionalTests/Query/GearsOfWarQuerySqlServerTest.cs]: some tests are now strongly typed to scalar/nullable scalar - those methods are not converted yet. Other tests use CollectionSorter, which for now is dynamic based - will uncomment all those tests once their requirements are met

File: test/EFCore.Specification.Tests/Query/GroupByQueryTestBase.cs (hunk 2 above)
Comment (maumar) [reply test/EFCore.Specification.Tests/Query/GroupByQueryTestBase.cs]: >.<

**Pull request merged**